libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -552,13 +552,8 @@ TickType_t uxIdentifier = 0U;
 				FreeRTOS_printf( ( "prvPrepareLookup: name is too long ( %lu > %lu )\n",
 								   ( unsigned long ) xLength,
 								   ( unsigned long ) ipconfigDNS_CACHE_NAME_LENGTH ) );
-				xLengthOk = pdFALSE;
 			}
 		}
-		else
-		{
-			xLengthOk = pdFALSE;
-		}
```

Comment:
I think we should keep the `else` statement. It can be empty, but still we should keep it. (Due to MISRA requirement of having and `else` for every `if`).
```
else
{
    /* Do Nothing. */
}
```

Reply:
I haven't seen that MISRA rule yet. What I did see is the following, both from PC-Lint and Coverity:
~~~c
if( count > max_count )
{
	count  = max_count;
}
else if( count < min_count )
{
	count  = min_count;
}
/* an else clause is expected here. */
~~~
error 9013: (Note -- no 'else' at end of 'if ... else if' chain [MISRA 2012 Rule 15.7, required])

What I did see in the FreeRTOS kernel is this:
~~~c
    if( pcName[ x ] == ( char ) 0x00 )
    {
        break;
    }
    else
    {
        mtCOVERAGE_TEST_MARKER();
    }
~~~